doc/ssdmond/ssdmond-hld.md

```diff
@@ -0,0 +1,129 @@
+This section will therefore only go into detail about data collection of attributes mentioned in [section 2.1](#21-priority-0-attributes):
+
+#### **2.4.1 IO Reads/Writes**
+
+- grep `/proc/diskstats` for statistics about the SSD of interest
+- Read the 4th value for reads completed successfully and the 8th value for writes completed<sup>[2](#2-kernelorg-procdiskstats)</sup>
+
+#### **2.4.2 Reserved Blocks Count**
+
+- Examine the SMART data for attributes related to reserved blocks or over-provisioning.
+- The exact attribute name and number can vary depending on the SSD manufacturer and model.
+- Look for keywords like "Reserved Block Count," "Over Provisioning," or similar terms.
+
+    Here's an example of what the SMART data output might look like:
+
+    ```
+    ...
+    ID# ATTRIBUTE_NAME          FLAGS    VALUE WORST THRESH FAIL RAW_VALUE
+    ...
+    173 Wear_Leveling_Count     0x0032   100   100   000    Old_age   Always       -       123
+    192 Power-Off_Retract_Count 0x0032   100   100   000    Old_age   Always       -       456
+    ...
+
+    ```
+
+    In this example, the "Wear_Leveling_Count" attribute might be indicative of reserved blocks or over-provisioning. However, the specific attribute and its interpretation can vary, so we make the search term and ID configurable by our vendors while maintaining a default search term in the event that this value is left unconfigured.
+
+## **StateDB Schema**
+```
+; Defines information for the SSD in a device
+
+key                 = SSD_INFO                       ; This key is for information that does not change for the lifetime of the SSD
+
+; field              = value
+
+Temperature         = STRING                            ; Describes the operating temperature of the SSD                                        (Priority 0, Dynamic)
+io_reads            = INT                               ; Describes the total number of reads completed successfully from the SSD               (Priority 0, Dynamic)
+io_writes           = INT                               ; Describes the total number of writes completed on the SSD                             (Priority 0, Dynamic)
+reserve_blocks      = INT                               ; Describes the reserved blocks count of the SSD                                        (Priority 0, Dynamic)
+device_model        = STRING                            ; Describes the Vendor information of the SSD                                           (Priority 1, Static)
+serial              = STRING                            ; Describes the Serial number of the SSD                                                (Priority 1, Static)
+firmware            = STRING                            ; Describes the Firmware version of the SSD                                             (Priority 1, Static)
+health              = STRING                            ; Describes the overall health of the SSD                                               (Priority 1, Dynamic)
```

Comment:
@assrinivasan please define what is health?

Reply:
From my research, I can summarize it as an overall health score based on the SMART attributes expressed as a percentage value by SMART tools